In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import ttest_ind

import warnings
warnings.filterwarnings("ignore")

# 데이터 불러오기
picher_file_path = '../data/picher_stats_2017.csv'
batter_file_path = '../data/batter_stats_2017.csv'
picher = pd.read_csv(picher_file_path)
batter = pd.read_csv(batter_file_path)

# 팀 정보와 연봉 정보가 포함된 데이터프레임 생성
baseball = pd.concat([picher, batter])

# 한화와 SK 데이터만 추출
hanhwa = baseball.loc[baseball['팀명'] == '한화']
sk = baseball.loc[baseball['팀명'] == 'SK']

# 각 팀의 연봉 평균 계산
hanhwa_mean = hanhwa['연봉(2017)'].mean()
sk_mean = sk['연봉(2017)'].mean()

# t-test로 두 팀 간의 차이 검정
t_stat, p_val = ttest_ind(hanhwa['연봉(2017)'], sk['연봉(2017)'], equal_var=False)

print('한화 평균 연봉:', hanhwa_mean)
print('SK 평균 연봉:', sk_mean)
print('t-test 통계량:', t_stat)
print('p-value:', p_val)

한화 평균 연봉: 22253.846153846152
SK 평균 연봉: 17736.842105263157
t-test 통계량: 0.6624041217840793
p-value: 0.5099280465043418


문제 : 한화팀과 SK팀간의 2017년 연봉의 평균간에는 유의미한 차이가 존재하는지 t-test로 검정하기

투수와 타자의 데이터를 불러오고,
우리가 구하고자 하는 것은 한화와 SK의 연봉 평균이어서 타자와 투수의 데이터를 'baseball' 이라는 데이터프레임에 합친다.
그 중에서도 한화와 SK의 데이터를 추출하고, 평균값을 계산한다.
그 후 t-test와 p-value로 차이를 검정하였고, p-value가 0.5099로 나타나므로, 통계적으로 유의미한 차이가 없다고 해석하였다.
